src/Hl7.Fhir.Base/Serialization/FhirJsonConverter.cs

```diff
@@ -15,97 +15,95 @@
 using Hl7.Fhir.Utility;
 using System;
 using System.Collections.Generic;
-using System.Linq;
 using System.Reflection;
 using System.Text.Json;
 using System.Text.Json.Serialization;
-using JsonSerializer = Newtonsoft.Json.JsonSerializer;
 
 namespace Hl7.Fhir.Serialization
 {
     /// <summary>
     /// A converter factory to construct FhirJsonConverters for subclasses of <see cref="Base"/>.
     /// </summary>
-    public class FhirJsonConverterFactory : JsonConverterFactory
+    public class FhirJsonConverterFactory(ModelInspector inspector, FhirJsonPocoSerializerSettings serializerSettings, FhirJsonPocoDeserializerSettings deserializerSettings) : JsonConverterFactory
     {
-        private readonly ModelInspector _inspector;
-        internal Predicate<CodedException> IgnoreFilter;
-        private readonly FhirJsonPocoSerializerSettings? _serializerSettings;
-        private readonly FhirJsonPocoDeserializerSettings? _deserializerSettings;
+        internal PocoSerializationEngine? Engine { get; set; }
 
         public FhirJsonConverterFactory(
-            Assembly assembly,
-            FhirJsonPocoSerializerSettings? serializerSettings = null,
-            FhirJsonPocoDeserializerSettings? deserializerSettings = null,
-            Predicate<CodedException>? ignoreFilter = null) : this(ModelInspector.ForAssembly(assembly), serializerSettings, deserializerSettings, ignoreFilter)
+            Assembly assembly, FhirJsonPocoSerializerSettings serializerSettings, FhirJsonPocoDeserializerSettings deserializerSettings) : this(ModelInspector.ForAssembly(assembly), serializerSettings, deserializerSettings)
         {
             // Nothing
         }
 
-        public FhirJsonConverterFactory(
-            ModelInspector inspector,
-            FhirJsonPocoSerializerSettings? serializerSettings = null,
-            FhirJsonPocoDeserializerSettings? deserializerSettings = null,
-            Predicate<CodedException>? ignoreFilter = null)
+        internal void SetEnforcedErrors(IEnumerable<string> toEnforce)
+        {
+            Engine ??= (FhirSerializationEngineFactory.Strict(inspector, serializerSettings, deserializerSettings) as PocoSerializationEngine)!;
+            Engine.IgnoreFilter = Engine.IgnoreFilter.And(toEnforce.ToPredicate().Negate());            
+        }
+
+        internal void SetIgnoredErrors(IEnumerable<string> toIgnore)
+        {
+            Engine ??= (FhirSerializationEngineFactory.Strict(inspector, serializerSettings, deserializerSettings) as PocoSerializationEngine)!;
```

Comment:
Maybe this factory code should be in a small private function, instead of repeating it.